### PR TITLE
[display_list] allow applying opacity peephole to single glyph.

### PR DESCRIPTION
**Pull request by jonahwilliams**
A single glyph can be opacity peepholed, this is common in the case of Icons.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
FYI @flar , not sure if we have a standard harness for testing this sort of thing, or if you have cases somewhere for textblob I could expand to textframe.

**Comment by flar** [comment]
dl_rendering_unittests.cc has some "compare with/without opacity" kinds of tests, but they don't run on Impeller. Would it be possible to create an Impeller golden test using DLBuilder? I thought that we recently got the DL playground tests to run as goldens.

**Comment by flar** [comment]
Alternately, render it with an alpha color and render it with an alpha saveLayer and see if the two produce the same image. You can also check that the saveLayer DL reports `can_distribute_opacity()` as true on the DL object itself (as long as that one saveLayer is the only thing in it, it will indicate that it has correctly determined that the opacity can be peepholed)...

**Review by flar — APPROVED** [review]
Looks good. I found a code inconsistency while reviewing the changes here, but this change didn't introduce it so I filed a follow-up bug for when we have time for the maintenance...

File: display_list/dl_builder.cc [hunk 1]
```diff
@@ -1698,7 +1698,11 @@ void DisplayListBuilder::drawTextFrame(
     // they will protect overlapping glyphs from the effects of overdraw
     // so we must make the conservative assessment that this DL layer is
     // not compatible with group opacity inheritance.
-    UpdateLayerOpacityCompatibility(false);
+    // A single glyph can still have the opacity peephole applied (this is
+    // likely a glyph used as an Icon)
+    const bool is_single_glyph = text_frame->GetRunCount() == 1u &&
+                                 text_frame->GetRuns()[0].GetGlyphCount() == 1u;
+    UpdateLayerOpacityCompatibility(is_single_glyph);
```
[the remark this answers is not in the document]
Comment (flar) [reply display_list/dl_builder.cc:1705]: Also, of note, this doesn't cause a bug, it's just a systemic logic confusion that ends up doing the right thing anyway.

**Comment by jonahwilliams** [comment]
reason for revert: unexpected framework goldens https://flutter-gold.skia.org/search?issue=149634&crs=github&patchsets=4&corpus=flutter

**Pull request merged**